### ISSUE TITLE
False positive: Reference of variable passed as function use argument is overlooked

### ISSUE THREAD

**Issue by vertexvaar** [kalessil/phpinspectionsea#1734]
| Subject        | Details                                                                       |
| :------------- | :---------------------------------------------------------------------------- |
| Plugin         | Php Inspections (EA Extended) 4.0.6.3 |
| Language level | PHP 7.1                        |

Current behaviour
--
Following code is highlighted with the message `[EA] Assignment here is not making much sense.`
```php
function () {
        $function = null;
        return $function = function () use (&$function) {
            // Do things...
        };
    }
```
but actually `$function` is overwritten with the Closure before it will be returned which is required for the referenced variable to contain the Closure, too.

Expected behaviour
--
No error should be shown, when a referenced variable is overwritten.

Environment details
---
PhpStorm 2021.1.3 Preview
Build #PS-211.7442.14, built on May 20, 2021
Licensed to [...]
Subscription is active until September 4, 2021.
Runtime version: 11.0.11+9-b1341.57 amd64
VM: Dynamic Code Evolution 64-Bit Server VM by JetBrains s.r.o.
Linux 5.8.0-53-generic
GC: G1 Young Generation, G1 Old Generation
Memory: 1990M
Cores: 8
Registry: run.processes.with.pty=TRUE
Non-Bundled Plugins: Key Promoter X (2021.1.1), String Manipulation (8.15.203.000.3), name.kropp.intellij.makefile (211.6693.108), com.jetbrains.plugins.jade (211.6693.65), com.cedricziel.idea.typo3 (211.7.1), com.kalessil.phpStorm.phpInspectionsEA (4.0.6.3), de.espend.idea.php.annotation (8.0.0), fr.adrienbrault.idea.symfony2plugin (0.23.209), de.sgalinski.typoscript.enterprise.plugin.id (2.4.0), ru.adelf.idea.dotenv (2021.1)
Current Desktop: ubuntu:GNOME

**Comment by ktomk** [comment]
Yes, the _Useless return_ inspection highlights with a weak error this place and when you look into it, it does not apply.

The inspection plugin does not highlight as well the useless assignment the line above:

~~~
$function = null;
~~~

Adding parenthesis adds more clarity for the first to not trigger the weak warning and _may_ make the code more readable, too, but this is not meant to suggest in any way that the report would not be justified and is meant merely only informative:

~~~
    return ($function = function () use (&$function) {
        return $function;
    });
~~~

Your mileage may vary.

**Comment by ea-inspections-team** [comment]
Thanks for reporting @vertexvaar and @ktomk for your proposals ! The bug is fixed now.
